components/event-simulator/org.wso2.carbon.event.simulator.core/src/main/java/org/wso2/carbon/event/simulator/core/internal/ds/CarbonEventSimulator.java

```diff
@@ -926,6 +944,7 @@ public void run(){
                     }
                     event.setAttributeValues(attributeValues);
                     sendEvent(event);
+                    Thread.sleep(delayBetweenEventsInMilies);
```

Comment:
@ramindu90 

In this case, if the delay is set to 0 also, still the thread might go to sleep for some small time and come back. However, the user might be expecting the events to be pushed continuously if the delay is set to 0. This might not impact much in event simulator scenario, but if you are doing a performance test or similar, you might want to keep a separate path and handle the case of zero delay as a special case.

Also, have we validated for the delay being set to a negative value by the user?